### PR TITLE
feat(signals): take new auth in use for signals

### PR DESCRIPTION
**Pull request by drodil**
## Hey, I just made a Pull Request!

- fixed authentication for signals
- improved error logging

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: packages/backend/src/index.ts [hunk 1]
```diff
@@ -105,6 +106,7 @@ function makeCreateEnv(config: Config) {
     root.child({ type: 'plugin' }),
     eventsService,
   );
+  const userInfo = new UserInfoCompat();
```
Comment (Rugvip) [inline packages/backend/src/index.ts:109]: Let's skip this and the addition to the plugin environment. We're not wiring up these services for the old backend system

File: packages/backend-common/api-report.md [hunk 2]
```diff
@@ -853,4 +855,10 @@ export function useHotCleanup(
 
 // @public @deprecated
 export function useHotMemoize<T>(_module: NodeModule, valueFactory: () => T): T;
+
+// @public (undocumented)
+export class UserInfoCompat implements UserInfoService {
```
Comment (Rugvip) [inline packages/backend-common/api-report.md:860]: We don't wanna export this directly, the compat helper is the way to get access to it for now

File: plugins/signals-backend/src/service/router.test.ts [hunk 3]
```diff
@@ -38,6 +39,10 @@ const discovery: jest.Mocked<PluginEndpointDiscovery> = {
   getExternalBaseUrl: jest.fn(),
 };
 
+const userInfo: jest.Mocked<UserInfoService> = {
```
Comment (Rugvip) [inline plugins/signals-backend/src/service/router.test.ts:42]: You can use `mockServices` from backend-test-utils for this too

File: plugins/signals-backend/src/service/router.ts [hunk 4]
```diff
@@ -14,44 +14,118 @@
...
+    }
+
+    let userIdentity: BackstageUserInfo | undefined = undefined;
+
+    // Authentication token is passed in Sec-WebSocket-Protocol header as there
+    // is no other way to pass the token with plain websockets
+    try {
+      const token = request.headers['sec-websocket-protocol'];
+      if (token) {
+        const credentials = await httpAuth.credentials(
```
Comment (Rugvip) [inline plugins/signals-backend/src/service/router.ts:91]: Use the plain `AuthService` here instead, it's lower level and lets you hand over a token directly. If you want to limit to just users you can use the `isPrincipal` method.

File: plugins/signals-backend/src/service/router.ts [hunk 5]
```diff
@@ -14,44 +14,118 @@
...
+          {
+            ...request,
+            headers: { ...request.headers, authorization: `Bearer ${token}` },
+          } as Request<any, any, any, any, any>,
+          { allow: ['user', 'none'], allowLimitedAccess: true },
+        );
+
+        if (credentials.principal.type === 'user') {
```
Comment (Rugvip) [inline plugins/signals-backend/src/service/router.ts:99]: Would it be fine to limit this to only authenticated users at this point? We've got guest access for livable development with proper tokens now thanks to #22565

File: plugins/signals-backend/src/service/router.ts [hunk 6]
```diff
@@ -14,44 +14,118 @@
...
+  httpAuth?: HttpAuthService;
 }
 
 /** @public */
 export async function createRouter(
   options: RouterOptions,
 ): Promise<express.Router> {
-  const { logger, identity, discovery } = options;
+  const { logger, userInfo, discovery } = options;
+  const { httpAuth } = createLegacyAuthAdapters(options);
```
Comment (Rugvip) [inline plugins/signals-backend/src/service/router.ts:52]: userInfo gets returned from here too

**Review by Rugvip — CHANGES_REQUESTED** [review]
Nice! :grin:

File: packages/backend/src/index.ts (hunk 1 above)
Comment (drodil) [reply packages/backend/src/index.ts:109]: This means that `signals` will not work with the old backend system anymore as the `UserInfoService` is mandatory now?

File: plugins/signals-backend/src/service/router.test.ts (hunk 3 above)
Comment (drodil) [reply plugins/signals-backend/src/service/router.test.ts:42]: Yeah, though those were not exported yet

File: plugins/signals-backend/src/service/router.ts (hunk 5 above)
Comment (drodil) [reply plugins/signals-backend/src/service/router.ts:99]: Sure, works for me

File: plugins/signals-backend/src/service/router.ts (hunk 6 above)
Comment (drodil) [reply plugins/signals-backend/src/service/router.ts:52]: Ah, true, great!

File: plugins/signals-backend/src/service/router.ts (hunk 4 above)
Comment (drodil) [reply plugins/signals-backend/src/service/router.ts:91]: Yeah, will fix

File: plugins/signals-backend/src/service/router.ts [hunk 7]
```diff
@@ -14,44 +14,111 @@
...
+        }
+      }
+    } catch (e) {
+      logger.error('Failed to authenticate WebSocket connection', e);
+      socket.write(
+        'HTTP/1.1 401 Web Socket Protocol Handshake\r\n' +
+          'Upgrade: WebSocket\r\n' +
+          'Connection: Upgrade\r\n' +
+          '\r\n',
+      );
```
Comment (Rugvip) [inline plugins/signals-backend/src/service/router.ts:103]: 😅 , makes sense tho! 👍

**Review by Rugvip — APPROVED** [review]
:shipit: 🎉

**Pull request merged**